cloudstate/cloudstate.go

```diff
@@ -16,230 +16,92 @@
 package cloudstate
 
 import (
-	"context"
 	"errors"
 	"fmt"
 	"log"
 	"net"
 	"os"
-	"runtime"
 
+	"github.com/cloudstateio/go-support/cloudstate/crdt"
+	"github.com/cloudstateio/go-support/cloudstate/discovery"
+	"github.com/cloudstateio/go-support/cloudstate/entity"
+	"github.com/cloudstateio/go-support/cloudstate/eventsourced"
 	"github.com/cloudstateio/go-support/cloudstate/protocol"
-	"github.com/golang/protobuf/descriptor"
-	"github.com/golang/protobuf/proto"
-	filedescr "github.com/golang/protobuf/protoc-gen-go/descriptor"
-	"github.com/golang/protobuf/ptypes/empty"
 	"google.golang.org/grpc"
 )
 
-const (
-	SupportLibraryVersion = "0.1.1"
-	SupportLibraryName    = "cloudstate-go-support"
-)
-
-// CloudState is an instance of a CloudState User Function
+// CloudState is an instance of a Cloudstate User Function.
```

Comment:
CloudState is an instance of a CloudState User Function.?

Reply:
nice catch; do you refer to the word casing or the statement itself?

Reply:
Ok, this time `cloudstate.CloudState` is the Go type representing a Cloudstate user function instance. As Cloudstate, with lower case state, is the marketing name, it felt strange to name Go types using a marketing name and not CamelCase naming the type.

Actually we could name `cloudstate.CloudState` `cloudstate.Server` which would escape us from the unholy duplication of naming a type respecting the package to be part of the type named. I thought about that before, but did not yet was convinced to do it.

Reply:
Perfect!